### ISSUE TITLE
HttpListener Read_FullLengthAsynchronous_Success test failed in CI

### ISSUE THREAD

**Issue by stephentoub** [dotnet/corefx#18376]
https://ci.dot.net/job/dotnet_corefx/job/master/job/windows_nt_debug_prtest/6278/consoleText
```
System.Net.Tests.HttpRequestStreamTests.Read_FullLengthAsynchronous_Success(transferEncodingChunked: True, text: \"\") [FAIL]
        System.Net.Http.HttpRequestException : An error occurred while sending the request.
        ---- System.Net.Http.WinHttpException : The server returned an invalid or unrecognized response
        Stack Trace:
              at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
              at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
              at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
              at System.Runtime.CompilerServices.ConfiguredTaskAwaitable`1.ConfiguredTaskAwaiter.GetResult()
           D:\j\workspace\windows_nt_de---4526f5ff\src\System.Net.Http\src\System\Net\Http\HttpClient.cs(462,0): at System.Net.Http.HttpClient.<FinishSendAsyncBuffered>d__58.MoveNext()
           --- End of stack trace from previous location where exception was thrown ---
              at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
              at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
              at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
              at System.Runtime.CompilerServices.TaskAwaiter`1.GetResult()
           D:\j\workspace\windows_nt_de---4526f5ff\src\System.Net.HttpListener\tests\HttpRequestStreamTests.cs(74,0): at System.Net.Tests.HttpRequestStreamTests.<Read_FullLengthAsynchronous_Success>d__5.MoveNext()
           --- End of stack trace from previous location where exception was thrown ---
              at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
              at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
              at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
           --- End of stack trace from previous location where exception was thrown ---
              at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
              at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
              at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
           --- End of stack trace from previous location where exception was thrown ---
              at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
              at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
              at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
           ----- Inner Stack Trace -----
              at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
           D:\j\workspace\windows_nt_de---4526f5ff\src\Common\src\System\Threading\Tasks\RendezvousAwaitable.cs(62,0): at System.Threading.Tasks.RendezvousAwaitable`1.GetResult()
           D:\j\workspace\windows_nt_de---4526f5ff\src\System.Net.Http.WinHttpHandler\src\System\Net\Http\WinHttpHandler.cs(863,0): at System.Net.Http.WinHttpHandler.<StartRequest>d__105.MoveNext()
```

**Comment by davidsh** [comment]
Changed title from WinHttpHandler.

This test is about HttpListener.

**Comment by Jiayili1** [comment]
This issue is repro on core tests in build 20170418.01, detail: https://mc.dot.net/#/product/netcore/master/source/official~2Fcorefx~2Fmaster~2F/type/test~2Ffunctional~2Fcli~2F/build/20170418.01/workItem/System.Net.HttpListener.Tests/analysis/xunit/System.Net.Tests.HttpRequestStreamTests~2FRead_FullLengthAsynchronous_Success(transferEncodingChunked:%20True,%20text:%20%5C%22%5C%22)

**Comment by karelz** [comment]
@steveharter can you please include this bug also in your test run in a loop. We want to find out its frequency and if there is a repro.

**Comment by hughbe** [comment]
Another instance in #18950. Looks like this only happens with chunked content and the empty string?

**Comment by steveharter** [comment]
This does repro on offline CI machines, around 1 in 700 runs on Windows 2008.

However the related synchronous test Read_FullLengthSynchronous_Success fails around 1% of the time on 2008. It also only fails for chunked + empty string. Issue for that is https://github.com/dotnet/corefx/issues/18869

**Comment by karelz** [comment]
So there is repro and the Jenkins CI frequency is 1/week so far - keeping it in 2.0 for now.

**Comment by Jiayili1** [comment]
This issue is repro on build 20170425.01, detail: https://mc.dot.net/#/product/netcore/master/source/official~2Fcorefx~2Fmaster~2F/type/test~2Ffunctional~2Fportable~2Fcli~2F/build/20170425.01/workItem/System.Net.HttpListener.Tests/analysis/xunit/System.Net.Tests.HttpRequestStreamTests~2FRead_FullLengthAsynchronous_Success(transferEncodingChunked:%20True,%20text:%20%5C%22%5C%22)

**Comment by tmat** [comment]
https://ci.dot.net/job/dotnet_corefx/job/master/job/windows_nt_debug_prtest/7218/

**Comment by Jiayili1** [comment]
This issue is repro on portable core tests, detail: https://mc.dot.net/#/product/netcore/master/source/official~2Fcorefx~2Fmaster~2F/type/test~2Ffunctional~2Fportable~2Fcli~2F/build/20170502.01/workItem/System.Net.HttpListener.Tests/analysis/xunit/System.Net.Tests.HttpRequestStreamTests~2FRead_FullLengthAsynchronous_Success(transferEncodingChunked:%20True,%20text:%20%5C%22%5C%22)

**Comment by davidsh** [comment]
Will probably be fixed with PR #19251

**Comment by davidsh** [comment]
Fixed with PR https://github.com/dotnet/corefx/issues/19251
